pkg/synthetictests/operators.go

```diff
@@ -73,6 +73,82 @@ func testOperatorStateTransitions(events monitorapi.Intervals, conditionTypes []
 	return ret
 }
 
+func testOperatorOSUpdateStaged(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	testName := "[bz-Machine Config Operator] Nodes should reach OSUpdateStaged in a timely fashion"
+	success := &junitapi.JUnitTestCase{Name: testName}
+
+	type startedStaged struct {
+		// OSUpdateStarted is the event Reason emitted by the machine config operator when a node begins extracting
+		// it's OS content.
+		OSUpdateStarted time.Time
+		// OSUpdateStaged is the event Reason emitted by the machine config operator when a node has extracted it's
+		// OS content and is ready to begin the update. For the purposes of this test, we're looking for how long it
+		// took from Started -> Staged to try to identify disk i/o problems that may be occurring.
+		OSUpdateStaged time.Time
+	}
+
+	// Scan all OSUpdateStarted and OSUpdateStaged events, sort by node.
+	nodeOSUpdateTimes := map[string]startedStaged{}
+	for _, e := range events {
+		if strings.Contains(e.Message, "reason/OSUpdateStarted") {
+			// locator will be of the form: node/ci-op-j34hmfqt-253f3-cq852-master-1
+			_, ok := nodeOSUpdateTimes[e.Locator]
+			if !ok {
+				nodeOSUpdateTimes[e.Locator] = startedStaged{}
+			}
+			// for this type of event, the from/to time are identical as this is a point in time event.
+			ss := nodeOSUpdateTimes[e.Locator]
+			ss.OSUpdateStarted = e.From
+		} else if strings.Contains(e.Message, "reason/OSUpdateStaged") {
+			// locator will be of the form: node/ci-op-j34hmfqt-253f3-cq852-master-1
+			_, ok := nodeOSUpdateTimes[e.Locator]
+			if !ok {
+				nodeOSUpdateTimes[e.Locator] = startedStaged{}
+			}
+			// for this type of event, the from/to time are identical as this is a point in time event.
+			ss := nodeOSUpdateTimes[e.Locator]
+			ss.OSUpdateStaged = e.From
+		}
+	}
+
+	// Iterate the data we assembled looking for any nodes with an excessive time between Started/Staged, or those
+	// missing a Staged
+	slowStageMessages := []string{}
+	var failTest bool // set true if we see anything over 10 minutes, our failure threshold
+	for node, ss := range nodeOSUpdateTimes {
+		if ss.OSUpdateStaged.IsZero() {
+			slowStageMessages = append(slowStageMessages, fmt.Sprintf("%s OSUpdateStarted at %s, did not make it to OSUpdateStaged", node, ss.OSUpdateStarted))
+			failTest = true
+		} else if ss.OSUpdateStaged.Sub(ss.OSUpdateStarted) > 5*time.Minute {
```

Comment:
possible worth an additional case here for `ss.OSUpdateStarted.IsZero()` for "best-effort event submission must have dropped the started event, so I dunno how long this node too to get from there to staged".  I guess I'd expect that to be a flake, but should be rare enough that if we decide to fail on it the number of runs that fail as a result should be small.